src/pkg/services/m365/api/groups.go

```diff
@@ -111,25 +113,28 @@ func getGroups(
 	return groups, el.Failure()
 }
 
-func FetchOnlyTeams(ctx context.Context, g models.Groupable) bool {
+func FetchOnlyTeams(ctx context.Context, groups []models.Groupable) []models.Groupable {
 	log := logger.Ctx(ctx)
-
-	if g.GetAdditionalData()[ResourceProvisioningOptions] != nil {
-		val, _ := tform.AnyValueToT[[]any](ResourceProvisioningOptions, g.GetAdditionalData())
-		for _, v := range val {
-			s, err := str.AnyToString(v)
-			if err != nil {
-				log.Debug("could not be converted to string value: ", ResourceProvisioningOptions)
-				return false
-			}
-
-			if s == teamsAdditionalDataLabel {
-				return true
+	var teams []models.Groupable
+
+	for _, g := range groups {
+		if g.GetAdditionalData()[ResourceProvisioningOptions] != nil {
```

Comment:
```suggestion
		if g.GetAdditionalData()[ResourceProvisioningOptions] != nil {
		  continue
		}
		// the rest
```

Reply:
I am little confused here. We want to continue is there is some value in `g.GetAdditionalData()[ResourceProvisioningOptions]` ?